### PR TITLE
Meme preview

### PR DESCRIPTION
**Pull request by spanjshiri**
Added an image preview that overlays text to the picture in realtime.

### CODE REVIEW

**Review by jphilly11 — COMMENTED** [review]
We might want to change our naming conventions for the different div's and tag's so that it is less confusing

**Review by jphilly11 — APPROVED** [review]

**Review by jphilly11 — DISMISSED** [review]

File: ui/src/App.tsx [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
+    super(props);
+    this.state = {topText: '', bottomText: ''}
+  }
+
+  getTopText = (text: String) => {
+    let newState = this.state;
+    newState = {
+      topText: text,
+      bottomText: this.state.bottomText
+    }
```
Comment (mchughmk) [inline ui/src/App.tsx]: Much of this is unnecessary.  If only changing the `topText` value, that's all you need to send to `setState`.  No need to send both values.

File: ui/src/App.tsx [hunk 2]
```diff
@@ -1,19 +1,45 @@
...
+    this.setState(newState); 
+  }
+
+  getBottomText = (text: String) => {
+    let newState = this.state;
+    newState = {
+      topText: this.state.topText,
+      bottomText: text
+    }
```
Comment (mchughmk) [inline ui/src/App.tsx]: Same comment as above.

File: ui/src/TextUpload.tsx [hunk 3]
```diff
@@ -1,26 +1,21 @@
...
 
-type TextProps = {}
-
-type TextState = {
-    topText: String,
-    bottomText: String,
-    errMsg: String
-}
-
-class TextUpload extends React.Component<TextProps, TextState> {
```
Comment (mchughmk) [inline ui/src/TextUpload.tsx]: Why did you remove these types?

**Review by mchughmk — CHANGES_REQUESTED** [review]

File: ui/src/App.tsx [hunk 4]
```diff
@@ -1,19 +1,43 @@
...
+    this.state = {topText: '', bottomText: ''}
+  }
+
+  getTopText = (text: String) => {
+    this.setState({topText: text});
+  }
+
+  getBottomText = (text: String) => {
+    this.setState({bottomText: text});
+  }
```
Comment (mchughmk) [inline ui/src/App.tsx]: These function names are misleading.  They say `get`, but they actually `set`.

File: ui/src/TextUpload.tsx [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
 
-type TextProps = {}
+type TextProps = {
+    sendTopText: CallableFunction,
+    sendBottomText: CallableFunction
+}
 
 type TextState = {
     topText: String,
     bottomText: String,
```
Comment (mchughmk) [inline ui/src/TextUpload.tsx:12]: Do we need to maintain these in this state anymore?  We've moved the state to the parent component and use prop functions to pass the data up.  These aren't used in this component anywhere.

**Review by mchughmk — CHANGES_REQUESTED** [review]

File: ui/src/TextUpload.tsx (hunk 5 above)
Comment (spanjshiri) [reply ui/src/TextUpload.tsx:12]: I am using the topText and bottomText state when I send it off in the post request. Is this not necessary?

Comment (mchughmk) [reply ui/src/TextUpload.tsx:12]: Oh, that makes sense, the `POST` does get called from this component, doesn't it?

**Review by mchughmk — DISMISSED** [review]

File: api/src/main/java/com/exelaration/abstractmemery/domains/AuditModel.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package com.exelaration.abstractmemery.domains;
```
Comment (KatherynRizek) [inline api/src/main/java/com/exelaration/abstractmemery/domains/AuditModel.java]: Should this file be removed? I thought I took it out previously - I'm not sure how it got back in

Comment (spanjshiri) [reply api/src/main/java/com/exelaration/abstractmemery/domains/AuditModel.java]: Idk how either, but nice catch I'll delete it

**Review by KatherynRizek — APPROVED** [review]

**Review by mchughmk — APPROVED** [review]
